ha/setup/post_disruptive_upgrade.py

```diff
@@ -0,0 +1,77 @@
+# coding: utf8
+
+import os
+from shutil import copystat
+
+from cortx.utils.log import Log
+from ha.execute import SimpleCommand
+from ha.const import (BACKUP_DEST_DIR_CONF, CONFIG_DIR, RA_LOG_DIR, \
+                        LIST_PCS_RESOURCES, CHECK_PCS_STANDBY_MODE, \
+                        CLUSTER_STANDBY_UNSTANDBY_TIMEOUT, PCS_CLUSTER_STANDBY, \
+                        PCS_CLUSTER_UNSTANDBY)
+from ha.setup.create_pacemaker_resources import create_all_resources
+
+
+def check_for_any_resource_presence() -> None:
+    '''Check if any resources are already present in a cluster.
+       if yes, means, pre-upgrade steps failed. hence exit'''
+
+    Log.info('Check for any resource presence in a cluster')
+
+    resource_list = SimpleCommand().run_cmd(LIST_PCS_RESOURCES)
+
+    resource_list[0].split('\n')
+
+    if resource_list:
+        Log.error('Some resources are already present. \
+                    Perform Upgrade again')
+        return
+
+def is_cluster_standby_on() -> None:
+    '''Check if cluster is in standby mode. If not, make standby mode ON'''
+
+    Log.info('Check cluster is in standby mode')
+    value = SimpleCommand().run_cmd(CHECK_PCS_STANDBY_MODE)
+
+    value = value[0].split(' ')[3].strip('\n').split('=')
+
+    if value[1].lower() != 'on':
+        Log.error('cluster is not in standby mode. Making standby mode ON')
+        standby_cmd = PCS_CLUSTER_STANDBY + f' --wait={CLUSTER_STANDBY_UNSTANDBY_TIMEOUT}'
+        SimpleCommand().run_cmd(standby_cmd)
+        Log.info('cluster standby mode on success')
+
+def restore_consul_backup():
+
+    Log.info('Restoring the consul backup')
+
+def restore_config() -> None:
```

Comment:
The old and new config needs to be merged, restore is just one scenario.

Reply:
Handled in a different way now. Please check.